### ISSUE TITLE
Bug in U2 gate parameter initialization

### ISSUE THREAD

**Issue by yezhuoyang** [mit-han-lab/torchquantum#232]
This is the code I use that has some bugs related with parameter initialization(In dev branch):

```Python
import torchquantum as tq
import torchquantum.functional as tqf

qdev = tq.QuantumDevice(n_wires=2, bsz=5, device="cpu", record_op=True)  # use device='cuda' for GPU

# use tqf
tq.u2(qdev, [0], [0.1, 0.2])

# print the current state (dynamic computation graph supported)
print(qdev)
```
U2 gate is the rotation gate with two parameters 
[U2 gate in qiskit](https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.U2Gate)

Running the above example will generate the following error message:

![image](https://github.com/mit-han-lab/torchquantum/assets/34168020/bdf27787-5e28-49ae-b7a6-5813abd9b585)

The error is caused by some strange manipulation on the dimension of the tensor, I don't understand why we have to use some squeeze and unsqueeze here.

![image](https://github.com/mit-han-lab/torchquantum/assets/34168020/e4af80a0-1a48-42d2-b07a-8506ee9c00a0)

It seems that many other gates with more than one parameters have similar issue. We should set the standard of how should user pass  the parameters in. And perhaps more code comments in the above part can help developer to understand the convention of tensor dimensions.

**Comment by Hanrui-Wang** [comment]
This is because we should use `tq.u2(qdev, [0],[[0.1, 0.2]])` for the computation with bsz>1

Let's fix it to make sure it runs with [0.1, 0.2] also

**Comment by tring27** [comment]
I think it should be ?
`if params.dim() == 1:
    params = params.unsqueeze(0)`
    
 this should make [0.1, 0.2] into [[0.1, 0.2]]
